bluesky_queueserver/server/tests/test_http_server.py

```diff
@@ -0,0 +1,185 @@
+import subprocess
+import sys
+import time as ttime
+
+import pytest
+import requests
+from xprocess import ProcessStarter
+
+import bluesky_queueserver.server.server as bqss
+from bluesky_queueserver.manager.tests.test_general import \
+    re_manager  # noqa F401
+from bluesky_queueserver.server.server import init_func
+
+SERVER_IP = '0.0.0.0'
+SERVER_PORT = '8080'
+
+
+@pytest.fixture
+def aiohttp_server(xprocess):
+    class Starter(ProcessStarter):
+        pattern = "Connected to ZeroMQ"
+        args = (f"{sys.executable} -m aiohttp.web -H {SERVER_IP} -P {SERVER_PORT} "
+                f"{bqss.__name__}:{init_func.__name__}").split()
+
+    xprocess.ensure("aiohttp_server", Starter)
+    # Clear the queue before the run:
+    subprocess.run('qserver -c clear_queue'.split())
+
+    yield
+
+    # Clear the queue after the run:
+    subprocess.run('qserver -c clear_queue'.split())
+    xprocess.getinfo("aiohttp_server").terminate()
+
+
+@pytest.fixture
+def add_plans_to_queue():
+    subprocess.run('qserver -c clear_queue'.split())
+    subprocess.call(["qserver", "-c", "add_to_queue", "-v",
+                     "{'name':'count', 'args':[['det1', 'det2']], 'kwargs':{'num':10, 'delay':1}}"])
+    subprocess.call(["qserver", "-c", "add_to_queue", "-v",
+                     "{'name':'count', 'args':[['det1', 'det2']]}"])
+    subprocess.call(["qserver", "-c", "add_to_queue", "-v",
+                     "{'name':'count', 'args':[['det1', 'det2']]}"])
+
+    yield
+
+    subprocess.run('qserver -c clear_queue'.split())
+
```

Comment:
```suggestion
    try:
        p.wait(5)
    except subprocess.TimeoutExpired:
        # The manager is not responsive, so just kill the process.
        p.kill()
        assert False, "RE Manager failed to stop"
```
I would put this code here. Sometimes the manager is frozen and don't stop. This means there is a bug and it will be indicated as an error in test results. This also kills the process so that other tests could use the manager.